File: frontend/src/api/ruleActionLog.ts
import { request } from './request';

export type ActionMode = 'NOTICE' | 'SOFT' | 'BLOCK';

export interface RuleActionLog {
  log_id: string;
  rule_code: string;
  rule_version: string;
  patient_id: string;
  encounter_id: string;
  order_id: string;
  action_mode: ActionMode;
  decision: 'CONTINUE' | 'MODIFY' | 'CANCEL';
  decision_by: string;
  decision_time: string;
  reason: string;
  informed_consent: boolean;
  family_notified: boolean;
  trace_id: string;
  tenant_id: string;
  group_code: string;
  hospital_code: string;
  campus_code: string;
  site_code: string;
  department_code: string;
  scope_level: string;
  scope_code: string;
  org_source: string;
  created_time: string;
}

export interface RecordDecisionRequest {
  rule_code: string;
  rule_version: string;
  patient_id: string;
  encounter_id: string;
  order_id: string;
  action_mode: ActionMode;
  decision: 'CONTINUE' | 'MODIFY' | 'CANCEL';
  decision_by: string;
  reason?: string;
  informed_consent?: boolean;
  family_notified?: boolean;
}

/**
 * 记录用户决策
 */
export async function recordDecision(request: RecordDecisionRequest): Promise<RuleActionLog> {
  const response = await request.post<RuleActionLog>('/api/rule-action-logs', request);
  return response.data;
}

/**
 * 查询决策日志列表
 */
export async function fetchActionLogs(params: {
  patient_id?: string;
  encounter_id?: string;
  rule_code?: string;
  decision?: string;
  decision_by?: string;
  limit?: number;
}): Promise<RuleActionLog[]> {
  const response = await request.get<RuleActionLog[]>('/api/rule-action-logs', { params });
  return response.data;
}

/**
 * 根据ID获取决策日志
 */
export async function fetchActionLog(logId: string): Promise<RuleActionLog> {
  const response = await request.get<RuleActionLog>(`/api/rule-action-logs/${logId}`);
  return response.data;
}

/**
 * 根据患者查询决策日志
 */
export async function fetchActionLogsByPatient(
  patientId: string,
  encounterId?: string
): Promise<RuleActionLog[]> {
  const params = encounterId ? { encounter_id: encounterId } : {};
  const response = await request.get<RuleActionLog[]>(
    `/api/rule-action-logs/patient/${patientId}`,
    { params }
  );
  return response.data;
}

/**
 * 根据订单查询决策日志
 */
export async function fetchActionLogsByOrder(orderId: string): Promise<RuleActionLog[]> {
  const response = await request.get<RuleActionLog[]>(`/api/rule-action-logs/order/${orderId}`);
  return response.data;
}

File: medkernel-mvp/src/main/java/com/medkernel/rule/RuleActionLog.java
package com.medkernel.rule;

import java.time.OffsetDateTime;

/**
 * 规则动作日志实体
 * 记录规则命中后的用户决策，用于审计和药师审方
 */
public class RuleActionLog {
    private Long id;
    private String logId;
    private String ruleCode;
    private String ruleVersion;
    private String patientId;
    private String encounterId;
    private String orderId;
    private ActionMode actionMode;
    private String decision;  // CONTINUE / MODIFY / CANCEL
    private String decisionBy;
    private OffsetDateTime decisionTime;
    private String reason;
    private boolean informedConsent;
    private boolean familyNotified;
    private String traceId;
    private String tenantId;
    private String groupCode;
    private String hospitalCode;
    private String campusCode;
    private String siteCode;
    private String departmentCode;
    private String scopeLevel;
    private String scopeCode;
    private String orgSource;
    private OffsetDateTime createdTime;

    public RuleActionLog() {
    }

    // Getters and Setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getLogId() {
        return logId;
    }

    public void setLogId(String logId) {
        this.logId = logId;
    }

    public String getRuleCode() {
        return ruleCode;
    }

    public void setRuleCode(String ruleCode) {
        this.ruleCode = ruleCode;
    }

    public String getRuleVersion() {
        return ruleVersion;
    }

    public void setRuleVersion(String ruleVersion) {
        this.ruleVersion = ruleVersion;
    }

    public String getPatientId() {
        return patientId;
    }

    public void setPatientId(String patientId) {
        this.patientId = patientId;
    }

    public String getEncounterId() {
        return encounterId;
    }

    public void setEncounterId(String encounterId) {
        this.encounterId = encounterId;
    }

    public String getOrderId() {
        return orderId;
    }

    public void setOrderId(String orderId) {
        this.orderId = orderId;
    }

    public ActionMode getActionMode() {
        return actionMode;
    }

    public void setActionMode(ActionMode actionMode) {
        this.actionMode = actionMode;
    }

    public String getDecision() {
        return decision;
    }

    public void setDecision(String decision) {
        this.decision = decision;
    }

    public String getDecisionBy() {
        return decisionBy;
    }

    public void setDecisionBy(String decisionBy) {
        this.decisionBy = decisionBy;
    }

    public OffsetDateTime getDecisionTime() {
        return decisionTime;
    }

    public void setDecisionTime(OffsetDateTime decisionTime) {
        this.decisionTime = decisionTime;
    }

    public String getReason() {
        return reason;
    }

    public void setReason(String reason) {
        this.reason = reason;
    }

    public boolean isInformedConsent() {
        return informedConsent;
    }

    public void setInformedConsent(boolean informedConsent) {
        this.informedConsent = informedConsent;
    }

    public boolean isFamilyNotified() {
        return familyNotified;
    }

    public void setFamilyNotified(boolean familyNotified) {
        this.familyNotified = familyNotified;
    }

    public String getTraceId() {
        return traceId;
    }

    public void setTraceId(String traceId) {
        this.traceId = traceId;
    }

    public String getTenantId() {
        return tenantId;
    }

    public void setTenantId(String tenantId) {
        this.tenantId = tenantId;
    }

    public String getGroupCode() {
        return groupCode;
    }

    public void setGroupCode(String groupCode) {
        this.groupCode = groupCode;
    }

    public String getHospitalCode() {
        return hospitalCode;
    }

    public void setHospitalCode(String hospitalCode) {
        this.hospitalCode = hospitalCode;
    }

    public String getCampusCode() {
        return campusCode;
    }

    public void setCampusCode(String campusCode) {
        this.campusCode = campusCode;
    }

    public String getSiteCode() {
        return siteCode;
    }

    public void setSiteCode(String siteCode) {
        this.siteCode = siteCode;
    }

    public String getDepartmentCode() {
        return departmentCode;
    }

    public void setDepartmentCode(String departmentCode) {
        this.departmentCode = departmentCode;
    }

    public String getScopeLevel() {
        return scopeLevel;
    }

    public void setScopeLevel(String scopeLevel) {
        this.scopeLevel = scopeLevel;
    }

    public String getScopeCode() {
        return scopeCode;
    }

    public void setScopeCode(String scopeCode) {
        this.scopeCode = scopeCode;
    }

    public String getOrgSource() {
        return orgSource;
    }

    public void setOrgSource(String orgSource) {
        this.orgSource = orgSource;
    }

    public OffsetDateTime getCreatedTime() {
        return createdTime;
    }

    public void setCreatedTime(OffsetDateTime createdTime) {
        this.createdTime = createdTime;
    }
}